valhalla/jawn/src/controllers/public/traceloopController.ts

```diff
@@ -0,0 +1,29 @@
+import {
+  Controller,
+  Request,
+  Route,
+  Post,
+  Tags,
+  Security,
+} from "tsoa";
+import type { JawnAuthenticatedRequest } from "../../types/request";
+import { OTELTrace, TraceManager } from "../../managers/traceManager";
+
+@Route("v1/trace")
+@Tags("Trace")
+@Security("api_key")
+export class TraceController extends Controller {
+  @Post("log")
+  public async logTrace(@Request() request: JawnAuthenticatedRequest) {
+    const traceManager = new TraceManager();
+    const trace = request.body as OTELTrace;
```

Comment:
Why is this not passed into the function as a `@body` parameter.

Doing that way will allow tsoa to typecheck the Body.